src/patternfly/components/Switch/switch.scss

```diff
@@ -21,31 +22,33 @@
   --pf-c-switch__toggle--BorderRadius: var(--pf-c-switch--Height);
   --pf-c-switch__toggle--before--Top: calc((var(--pf-c-switch--Height) - var(--pf-c-switch__toggle--before--Height)) / 2);
   --pf-c-switch__toggle--before--Left: var(--pf-c-switch__toggle--before--Top);
-  --pf-c-switch__toggle--before--Width: calc(var(--pf-c-switch__label--FontSize) - #{pf-size-prem(2px)});
+  --pf-c-switch__toggle--before--Width: calc(var(--pf-c-switch--FontSize) - var(--pf-c-switch__toggle-icon--Offset));
   --pf-c-switch__toggle--before--Height: var(--pf-c-switch__toggle--before--Width);
   --pf-c-switch__toggle--before--BackgroundColor: var(--pf-global--BackgroundColor--100);
   --pf-c-switch__toggle--before--BorderRadius: var(--pf-global--BorderRadius--lg);
   --pf-c-switch__toggle--before--BoxShadow: var(--pf-global--BoxShadow--md);
   --pf-c-switch__toggle--before--Transition: transform .25s ease 0s, width .25s ease 0s, left .25s ease 0s;
 
   // Switch icon
-  --pf-c-switch__toggle-icon--FontSize: var(--pf-global--icon--FontSize--sm);
+  --pf-c-switch__toggle-icon--FontSize: calc(var(--pf-c-switch--FontSize) * .625);
```

Comment:
Since we don't normally do this, can you add a comment here explaining why we don't use an `--icon--FontSize` var b/c this allows the icon to scale?